airmozilla/popcorn/views.py

```diff
@@ -106,6 +113,60 @@ def popcorn_data(request):
         }
 
 
+# Note that this view is publically available.
+# That means we can't trust the content but we can take it as a hint.
+@csrf_exempt
+@require_POST
+def vidly_webhook(request):
+    if not request.POST.get('xml'):
+        return http.HttpResponseBadRequest("no 'xml'")
+
+    xml_string = request.POST['xml'].strip()
+    try:
+        struct = xmltodict.parse(xml_string)
+    except ExpatError:
+        return http.HttpResponseBadRequest("Bad 'xml'")
+
+    try:
+        task = struct['Response']['Result']['Task']
+        migrate_submission(task)
+    except KeyError:
+        # If it doesn't have a "Result" or "Task", it was just a notification
+        # that the media was added.
```

Comment:
This is misleading. A `KeyError` could happen on line 131 and/or line 132. You just don't know. It's bad to put too broad `try:except` blocks on things. What if you have a bug in `migrate_submission()` that raises a `KeyError` sometimes then we'd want that to crash and burn so we can get alerted and fix it.